bindings/c/test/mako/mako.c

```diff
@@ -37,10 +42,46 @@
     if ((fdb_future_block_until_ready(_f)) != 0) {                             \
       fprintf(stderr, "ERROR: fdb_future_block_until_ready failed at %s:%d\n", \
               __FILE__, __LINE__);                                             \
-      goto FDB_FAIL;                                                           \
+      goto failExit;                                                           \
     }                                                                          \
   } while (0)
 
+#define fdb_wait_and_handle_error(_func, _f, _t)			\
+  do {									\
+    int err = wait_future(_f);						\
+    if (err) {								\
+      int err2;								\
+      if ((err != 1020 /* not_committed */) &&				\
+	  (err != 1021 /* commit_unknown_result */)) {			\
+	fprintf(stderr, "ERROR: Error %s (%d) occured at %s\n",		\
+		  #_func, err, fdb_get_error(err));			\
+      } else {								\
+	fprintf(annoyme, "ERROR: Error %s (%d) occured at %s\n",	\
+		  #_func, err, fdb_get_error(err));			\
+      }									\
+      fdb_future_destroy(_f);						\
+      _f = fdb_transaction_on_error(_t, err);				\
+      /* this will return the original error for non-retryable errors */	\
+      err2 = wait_future(_f);						\
+      fdb_future_destroy(_f);						\
+      if (err2) {							\
+        /* unretryable error */						\
+	fprintf(stderr,							\
+		"ERROR: fdb_transaction_on_error returned %d at %s:%d\n", \
+		err2, __FILE__, __LINE__);				\
+	if (err2 == 1020 /* not_committed */) {				\
```

Comment:
`fdb_transaction_on_error` will never return a `not_committed` error - so this if block can be removed.

Reply:
I believe this error could be returned if retry limits are used, though I'm not sure if that's happening in this code or not.

Reply:
Mako doesn't set a retry limit. But I guess we could leave this in just to make sure this is handled if this option is used in a later version...

Reply:
I just pushed to address this (by having a TODO comment 😆 )